hasher-matcher-actioner/hmalib/common/reactioner_models.py

```diff
@@ -0,0 +1,73 @@
+import typing as t
+
+from dataclasses import dataclass, fields
+
+from hmalib.models import MatchMessage, BankedSignal
+from hmalib.common.logging import get_logger
+from hmalib.common.actioner_models import ActionPerformer, ActionLabel
+from hmalib.aws_secrets import AWSSecrets
+
+from threatexchange.api import ThreatExchangeAPI
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class ReactActionPerformer(ActionPerformer):
+    @property
+    def reaction(self) -> str:
+        raise NotImplementedError
+
+    def perform_action(self, match_message: MatchMessage) -> None:
+        api_key = AWSSecrets.te_api_key()
+        api = ThreatExchangeAPI(api_key)
+
+        indicator_ids = {
+            dataset_match_details.banked_content_id
+            for dataset_match_details in match_message.matching_banked_signals
+            if dataset_match_details.bank_source == "te"
+        }
+
+        descriptor_ids = {
+            descriptor_id["id"]
+            for indicator_id in indicator_ids
+            for descriptor_id in api.get_threat_descriptors_from_indicator(indicator_id)
+        }
+
+        for id in descriptor_ids:
+            api.react_to_threat_descriptor(id, self.reaction)
+            logger.info("reacted %s to descriptor %s", self.reaction, id)
+
+
+class ReactInReviewActionPerformer(ReactActionPerformer):
+    reaction = "IN_REVIEW"
```

Comment:
Are these constants available in the python-threatexchange library? These seem very 'threatexchangey'. Should probably not be here.

Reply:
Not stored anywhere in python-threatexchange currently. I think these ActionPerformer classes are ThreatExchange specific so not entirely sure the issue